src/ripple/rpc/impl/LookupLedger.cpp

```diff
@@ -83,46 +64,66 @@ Status ledgerFromRequest (
         uint256 ledgerHash;
         if (! ledgerHash.SetHex (hashValue.asString ()))
             return {rpcINVALID_PARAMS, "ledgerHashMalformed"};
-           
+
         ledger = netOps.getLedgerByHash (ledgerHash);
+        if (ledger == nullptr)
+            return {rpcLGR_NOT_FOUND, "ledgerNotFound"};
     }
     else if (indexValue.isNumeric())
     {
         ledger = netOps.getLedgerBySeq (indexValue.asInt ());
-    }
-    else
-    {
-        auto const index = indexValue.asString ();
-        auto const isCurrent = index.empty() || index == "current";
-        if (isCurrent)
-            ledger = netOps.getCurrentLedger ();
-        else if (index == "closed")
-            ledger = netOps.getClosedLedger ();
-        else if (index == "validated")
-            ledger = netOps.getValidatedLedger ();
-        else
-            return {rpcINVALID_PARAMS, "ledgerIndexMalformed"};
-
         if (ledger == nullptr)
-            return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+            return {rpcLGR_NOT_FOUND, "ledgerNotFound"};
 
-        assert (ledger->isImmutable());
-        assert (ledger->isClosed() == !isCurrent);
+        if (ledger->getLedgerSeq () > netOps.getValidatedSeq () &&
+            isValidatedOld ())
+        {
+            return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+        }
     }
-
-    if (ledger == nullptr)
+    else
     {
-        auto cl = netOps.getCurrentLedger ();
-        if (cl == nullptr || ! synced (netOps, cl))
+        if (isValidatedOld ())
             return {rpcNO_NETWORK, "InsufficientNetworkMode"};
 
-        return {rpcLGR_NOT_FOUND, "ledgerNotFound"};
-    }
-
-    if (! synced (netOps, ledger))
-    {
-        ledger.reset ();
-        return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+        auto const index = indexValue.asString ();
+        if (index == "validated")
+        {
+            ledger = netOps.getValidatedLedger ();            
+            if (ledger == nullptr)
+                return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+
+            assert (ledger->isClosed ());
+        }
+        else
+        {
+            if (index.empty () || index == "current")
+            {
+                ledger = netOps.getCurrentLedger ();
+                assert (! ledger->isClosed ());
+            }
+            else if (index == "closed")
+            {
+                ledger = netOps.getClosedLedger ();
+                assert (ledger->isClosed ());
+            }
+            else
+            {
+                return {rpcINVALID_PARAMS, "ledgerIndexMalformed"};
+            }
+
+            if (ledger == nullptr)
+                return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+
+            if (ledger->getLedgerSeq () + minSequenceGap <
+                netOps.getValidatedSeq ())
+            {
+                ledger.reset ();
+                return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+            }
```

Comment:
This catches the startup edge case, where rippled will only a genesis ledger, or one shortly following it (ie. sequence number will be < 100 or so), and we get the latest validated ledger from the network, but the current / closed haven't been updated to that fact yet. The `minSequenceGap` is small so that this test can work for altnets, testnets, etc. Correct?